### DISCUSSION TITLE
append a varible to clipboard

### DISCUSSION THREAD

**Discussion by ghost** [xournalpp/xournalpp#4269] [Q&A] [answered]
hi, i´m working in a plugin but just got a little stuck, i want to assign a string to the clipboard. i wanted to use just the funtions from the API but seems that they won't do the job (as far as my known goes -im newbie to all this-). so if someone could guide me I'd appreciate it.

**Comment by rolandlo** [comment, marked as answer]
What OS do you use? You will most likely need a Lua module or a system tool to access the clipboard. For example on Windows you can use the `winapi` module, see this [stackoverflow question](https://stackoverflow.com/questions/47584734/is-there-a-way-to-set-the-clipboard-text-in-lua)

See also [this script](https://gist.github.com/AndrewHazelden/b9909520490624305183f7c8f77368a2) that uses [`xclip`](https://github.com/astrand/xclip). On Wayland you could use [`wl-clipboard`](https://github.com/bugaevc/wl-clipboard) instead of `xclip`.

**Reply by ghost** [reply]
oh i use windows 10. Anyways that script is what i was looking for. im so thankful
